app/models/watcher_notification_mailer.rb

```diff
@@ -49,25 +49,25 @@ def other_jobs_queued?(work_package)
                          "%NotificationJob%journal_id: #{work_package.journals.last.id}%").exists?
     end
 
-    def notify_about_watcher_added?(watcher, watcher_setter)
-      return false if notify_about_self_watching?(watcher, watcher_setter)
+    def notify_about_watcher_added?(user, watcher_setter, watchable)
```

Comment:
The method could be renamed now( and was misnamed before) as it does not check for whether the user wants to be informed about a watcher being added but, regardless of the action, about events on the watchable.